FreeBSD: Fix RLIMIT_FSIZE handling for block cloning

markjdb opened the pull request:
The change removes a redundant and erroneous RLIMIT_FSIZE check in the VOP_COPY_FILE_RANGE implementation.

### Motivation and Context
The change unbreaks the use of RLIMIT_FSIZE with programs that use copy_file_range(); this happens with at least some postfix scripts. It is FreeBSD-specific.

### Description
See the commit log message - when block cloning is enabled, the removed vn_rlimit_fsize() call may fail even for valid requests.

### How Has This Been Tested?
Manual testing using `limits -f` and a program which uses copy_file_range(2). In particular, I checked that the limit is still enforced correctly after this change.

### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:
- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [ ] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

mcmilk reviewed (approved):
We could maybe add a FreeBSD specific testcase for that in ZTS.
@markjdb - could you add this or share the script of the manual testings?

markjdb commented:
> We could maybe add a FreeBSD specific testcase for that in ZTS. @markjdb - could you add this or share the script of the manual testings?

Sure, I will try to add one. I'm not too familiar with the ZTS - why do we have both `tests/functional/bclone` and `tests/functional/block_cloning`?

tonyhutter commented:
@markjdb it looks like the FreeBSD builder died, so I just restarted Buildbot.  Would you mind re-pushing to kick off a new test?

tonyhutter reviewed (approved):
@markjdb unfortunately it looks like the Buildbot FreeBSD 13 builder is still down.  I manually ran the tests using our qemu-based github runners (still under development), and it passed on FreeBSD 13, 14, and 15: https://github.com/tonyhutter/zfs/actions/runs/10270871475

markjdb commented:
> @markjdb unfortunately it looks like the Buildbot FreeBSD 13 builder is still down. I manually ran the tests using our qemu-based github runners (still under development), and it passed on FreeBSD 13, 14, and 15: https://github.com/tonyhutter/zfs/actions/runs/10270871475

Thanks! I had verified the test case locally (and found an unrelated bug which I'll post a patch for this week), but only on FreeBSD as I'm not set up to test ZFS+Linux.

tonyhutter commented:
@markjdb I'm trying to merge this PR, but the code you added in dd77d8e06 was already in master via cbcb522439.  Can I just omit dd77d8e06 and merge the other two of your commits instead?:

db8c4fb8a FreeBSD: Fix RLIMIT_FSIZE handling for block cloning
4ec01dd32 ZTS: Add a test to verify that copy_file_range obeys RLIMIT_FSIZE

markjdb commented:
> @markjdb I'm trying to merge this PR, but the code you added in [dd77d8e](https://github.com/openzfs/zfs/commit/dd77d8e06f0f07f93c477e34dd206948b6e4d590) was already in master via [cbcb522](https://github.com/openzfs/zfs/commit/cbcb522439acf32a8345b6b3a489b5f9740a88f3). Can I just omit [dd77d8e](https://github.com/openzfs/zfs/commit/dd77d8e06f0f07f93c477e34dd206948b6e4d590) and merge the other two of your commits instead?:

Sure, of course. I just dropped that commit and rebased.

tonyhutter commented:
Merged as dea8fabf7 0ccd4b9d0
